stardis/radiation_field/opacities/opacities_solvers/util.py

```diff
@@ -20,33 +19,75 @@ def sigma_file(tracing_lambdas, temperatures, fpath):
     """
-
-    df = pd.read_csv(fpath, header=None, comment="#")
-    if np.isnan(df.loc[0, 0]):
-        file_wavelengths = np.array(df.loc[1:, 0])
-        file_temperatures = np.array(df.loc[0, 1:])
-        file_cross_sections = np.array(df.loc[1:, 1:]).T
-        fn = interp2d(file_wavelengths, file_temperatures, file_cross_sections)
-        sigmas = fn(tracing_lambdas, temperatures)
-    else:
-        file_wavelengths = np.array(df.loc[:, 0])
-        file_temperatures = np.array([0])
-        file_cross_sections = np.array(df.loc[:, 1])
-        fn = interp1d(
-            file_wavelengths,
-            file_cross_sections,
+    if (
+        spec == "H2plus_bf"
+    ):  # This section specifically ingests the Stancil 1994 h2_plus_bf_S1994.dat table found in data.
+        h2_plus_bf_table = pd.read_csv(fpath, delimiter="\s+", index_col=0, comment="#")
+        h2_plus_bf_table.replace({"-": "e-"}, regex=True, inplace=True)
+        h2_plus_bf_table = h2_plus_bf_table.astype(float)
+        file_wavelengths = (h2_plus_bf_table.index.values * u.nm).to(u.AA).value
+        file_temperatures = h2_plus_bf_table.columns.values.astype(int)
+        file_cross_sections = h2_plus_bf_table.to_numpy()
+
+        file_waves_mesh, file_temps_mesh = np.meshgrid(
+            file_wavelengths, file_temperatures, indexing="ij"
+        )
+        linear_interp_h2plus_bf = LinearNDInterpolator(
+            np.vstack([file_waves_mesh.ravel(), file_temps_mesh.ravel()]).T,
+            file_cross_sections.flatten(),
+        )
+        lambdas, temps = np.meshgrid(tracing_lambdas, temperatures)
+        sigmas = (
+            linear_interp_h2plus_bf(lambdas, temps) * 1e-18
+        )  # Scaling from Stancil 1994 table
+
+    elif (
+        spec == "Hminus_ff"
+    ):  # This section specifically ingests the Bell and Berrington 1987 h_minus_ff_B1987.dat table found in data.
+        h_minus_ff_table = pd.read_csv(fpath, delimiter="\s+", comment="#")
+        h_minus_ff_table.columns = h_minus_ff_table.columns.str.strip(",")
+
+        file_wavelengths = h_minus_ff_table[h_minus_ff_table.columns[0]].values
+        file_thetas = h_minus_ff_table.columns[1:].astype(float)
+        file_values = h_minus_ff_table.to_numpy()[:, 1:]
+
+        file_waves_mesh, file_thetas_mesh = np.meshgrid(
+            file_wavelengths, file_thetas, indexing="ij"
+        )
+        linear_interp_hminus_ff = LinearNDInterpolator(
+            np.vstack([file_waves_mesh.ravel(), file_thetas_mesh.ravel()]).T,
+            file_values.flatten(),
+        )
+        lambdas, thetas = np.meshgrid(tracing_lambdas, 5040 / temperatures)
+        sigmas = (
+            linear_interp_hminus_ff(lambdas, thetas)
+            * 1e-26  # Scaling value from Bell 1987 stable
+            * const.k_B.cgs.value
+            * temperatures[:, np.newaxis]
+        )
+    else:  # This is currently used to read h_minus_bf_W1979.dat, Wishart 1979
```

Comment:
It would be better to explicitly specify hminus_bf and raise an error if no source is requested.